lib/pavilion/plugins/commands/cat.py

```diff
@@ -44,28 +45,27 @@ def run(self, pav_cfg, args):
                           file=sys.stderr, color=output.RED)
             return errno.EEXIST
 
-        return print_file(job_dir/args.file)
+        return self.print_file(job_dir/args.file)
 
+    def print_file(self, file):
+        """Print the file at the given path.
+        :param path file: The path to the file to print.
+        """
 
-def print_file(file):
-    """Print the file at the given path.
-    :param path file: The path to the file to print.
-    """
+        try:
+            with file.open() as file:
+                while True:
+                    block = file.read(4096)
+                    if not block:
+                        break
+                    output.fprint(block, file=self.outfile, end="")
 
-    try:
-        with file.open() as file:
-            while True:
-                block = file.read(4096)
-                if not block:
-                    break
-                output.fprint(block, file=sys.stdout)
+        except IsADirectoryError:
```

Comment:
You should also catch OSError, IOError, and PermissionError. You can get them as a group, and then give a generic message like:
"Error reading file '{}': {}".format(file_path, err)